packages/babel-helpers/src/helpers/wrapRegExp.js

```diff
@@ -0,0 +1,65 @@
+/* @minVersion 7.2.6 */
+
+import setPrototypeOf from "setPrototypeOf";
+import inherits from "inherits";
+
+export default function _wrapRegExp() {
+  _wrapRegExp = function (re, groups) {
+    return new BabelRegExp(re, undefined, groups);
+  };
+
+  var _super = RegExp.prototype;
+  var _groups = new WeakMap();
+
+  function BabelRegExp(re, flags, groups) {
+    var _this = new RegExp(re, flags);
+    // if the regex is recreated with 'g' flag
+    _groups.set(_this, groups || _groups.get(re));
+    return setPrototypeOf(_this, BabelRegExp.prototype);
+  }
+  inherits(BabelRegExp, RegExp);
+
+  BabelRegExp.prototype.exec = function (str) {
+    var result = _super.exec.call(this, str);
+    if (result) result.groups = buildGroups(result, this);
+    return result;
+  };
+  BabelRegExp.prototype[Symbol.replace] = function (str, substitution) {
+    if (typeof substitution === "string") {
+      var groups = _groups.get(this);
+      return _super[Symbol.replace].call(
```

Comment:
nit: We can also hoist `_super[Symbol.replace]` if we want to golf a bit.

Reply:
Not caching it makes it possible to monkey-patch `RegExp.protototype[Symbol.replace]`, similarly to how its possible with native regexps.